ray-operator/controllers/ray/utils/validation.go

```diff
@@ -43,6 +43,26 @@ func ValidateRayClusterSpec(spec *rayv1.RayClusterSpec, annotations map[string]s
 		if len(workerGroup.Template.Spec.Containers) == 0 {
 			return fmt.Errorf("workerGroupSpec should have at least one container")
 		}
+		if workerGroup.MinReplicas == nil || workerGroup.MaxReplicas == nil {
+			return fmt.Errorf("worker group %s must set both minReplicas and maxReplicas", workerGroup.GroupName)
+		}
+		if *workerGroup.MinReplicas < 0 {
+			return fmt.Errorf("worker group %s has negative minReplicas %d", workerGroup.GroupName, *workerGroup.MinReplicas)
+		}
+		if *workerGroup.MaxReplicas < 0 {
+			return fmt.Errorf("worker group %s has negative maxReplicas %d", workerGroup.GroupName, *workerGroup.MaxReplicas)
+		}
+		if *workerGroup.MinReplicas > *workerGroup.MaxReplicas {
+			return fmt.Errorf("worker group %s has minReplicas %d greater than maxReplicas %d", workerGroup.GroupName, *workerGroup.MinReplicas, *workerGroup.MaxReplicas)
+		}
+		if workerGroup.Replicas != nil {
```

Comment:
This is not invalid. It's not uncommon for users to manually update `MinReplicas` / `MaxReplicas`. If Ray Autoscaler is enabled, `Replicas` will eventually fall into `MinReplicas` / `MaxReplicas`.

Reply:
Maybe we can validate that if Autoscaler is not enabled and `Replicas` is not between `MinReplicas` / `MaxReplicas`. This case is invalid.